pkg/asset/installconfig/google/google.go

```diff
@@ -0,0 +1,163 @@
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/defaults"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	survey "gopkg.in/AlecAivazis/survey.v1"
+
+	"github.com/openshift/installer/pkg/types/google/validation"
+)
+
+// Platform collects GCP-specific configuration.
+func Platform() (*google.Platform, error) {
+	longRegions := make([]string, 0, len(validation.Regions))
+	shortRegions := make([]string, 0, len(validation.Regions))
+	for id, location := range validation.Regions {
+		longRegions = append(longRegions, fmt.Sprintf("%s (%s)", id, location))
+		shortRegions = append(shortRegions, id)
+	}
+	regionTransform := survey.TransformString(func(s string) string {
+		return strings.SplitN(s, " ", 2)[0]
+	})
+
+	defaultRegion := "us-central1"
+	_, ok := validation.Regions[defaultRegion]
+	if !ok {
+		panic(fmt.Sprintf("installer bug: invalid default GCP region %q", defaultRegion))
+	}
+
+	var err error
+	// ssn, err := getSession()
+	// if err != nil {
+	// 	return nil, err
+	// }
+
+	// defaultRegionPointer := ssn.Config.Region
+	// if defaultRegionPointer != nil && *defaultRegionPointer != "" {
+	// 	_, ok := validation.Regions[*defaultRegionPointer]
+	// 	if ok {
+	// 		defaultRegion = *defaultRegionPointer
+	// 	} else {
+	// 		logrus.Warnf("Unrecognized GCP region %q, defaulting to %s", *defaultRegionPointer, defaultRegion)
+	// 	}
+	// }
+
+	sort.Strings(longRegions)
+	sort.Strings(shortRegions)
+
+	var region string
+	err = survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Select{
+				Message: "Region",
+				Help:    "The GCP region to be used for installation.",
+				Default: fmt.Sprintf("%s (%s)", defaultRegion, validation.Regions[defaultRegion]),
+				Options: longRegions,
+			},
+			Validate: survey.ComposeValidators(survey.Required, func(ans interface{}) error {
+				choice := regionTransform(ans).(string)
+				i := sort.SearchStrings(shortRegions, choice)
+				if i == len(shortRegions) || shortRegions[i] != choice {
+					return errors.Errorf("invalid region %q", choice)
+				}
+				return nil
+			}),
+			Transform: regionTransform,
+		},
+	}, &region)
+	if err != nil {
+		return nil, err
+	}
+
+	return &google.Platform{
+		Region: region,
+	}, nil
+}
+
+func getSession() (*session.Session, error) {
+	ssn := session.Must(session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	}))
+	ssn.Config.Credentials = credentials.NewChainCredentials([]credentials.Provider{
+		&credentials.EnvProvider{},
+		&credentials.SharedCredentialsProvider{},
+	})
+	_, err := ssn.Config.Credentials.Get()
+	if err == credentials.ErrNoValidProvidersFoundInChain {
+		err = getCredentials()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return ssn, nil
+}
+
+func getCredentials() error {
+	var keyID string
+	err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "AWS Access Key ID",
```

Comment:
Some AWS leftovers here.

Reply:
yeah, I don't take credentials yet.  Once we settle on behavior of the various operators on unrecognized clouds I'm going to go open a bunch of PRs /bugs and then push through more.